js/ui/pivot_grid/data_source.js

```diff
@@ -458,7 +458,9 @@ export default Class.inherit((function() {
                 } else {
                     fieldsDictionary[fieldKey] = mergedField = field;
                 }
-                extend(mergedField, { dataType: dataTypes[field.dataField] });
+                if(dataTypes[field.dataField]) {
+                    mergedField.dataType ||= dataTypes[field.dataField];
+                }
```

Comment:
Minor comment.

Let's make these lines more obvious:
```suggestion
                if(!mergedField.dataType && dataTypes[field.dataField]) {
                    mergedField.dataType = dataTypes[field.dataField];
                }
```